execute.go

```diff
@@ -17,14 +28,27 @@ func executeHelper(client *Client, method string, body []byte) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	resp, err := client.session.Do(req)
 	if err != nil {
 		return nil, err
 	}
 
-	respbody, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
+	respbody, rerr := io.ReadAll(resp.Body)
+	if rerr != nil {
+		return nil, rerr
+	}
+
+	// If we didnt get 20x response, unmarshal the error body to be able to format
```

Comment:
And can you add [this link](https://postgrest.org/en/stable/api.html#errors-and-http-status-codes) instead of comment lines.